### ISSUE TITLE
Specify minimum compiler version requirements

### ISSUE THREAD

**Issue by laanwj** [bitcoin/bitcoin#11251]
The 0.13.0 release notes mention, in the context of C++11 introduction:
```
Effectively this means GCC 4.7 or higher, or Clang 3.3 or higher.
```
We should document this minimum compiler requirement in the build guides somewhere.

I'm not entirely sure where though, we don't have a general build guide, only the per-OS ones.

**Comment by fanquake** [comment]
Can this be combined into the minimum required dependancies guide? 
Compiler requirements are listed in #8639, which is being turned into some actual docs in #10779.

**Comment by laanwj** [comment]
> Can this be combined into the minimum required dependancies guide?

Yes, I think that makes sense.
We could even add a check to `configure.ac`.

**Comment by laanwj** [comment]
Closing, this is sufficiently addressed by #10779 which is merged.
